nixos/modules/services/misc/navidrome.nix

```
@@ -0,0 +1,102 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.navidrome;
+  configFile = "/etc/navidrome/settings.json";
+  dataFolder = "/var/lib/navidrome";
+in {
+  options = {
+
+    services.navidrome = {
+      enable = mkEnableOption "Enable Navidrome music server and streamer";
+
+      settings = lib.mkOption {
+        type = with types; attrsOf (oneOf [ int str ]);
+        default = {};
+        example = literalExample ''
+          {
+            LogLevel = "INFO";
+            BaseURL = "/music";
+            ScanInterval = "10s";
+            TranscodingCacheSize = "15MB";
+            MusicFolder = "/Media/Music";
+          };
+        '';
+        description = ''
+          Configuration for Navidrome, see <link xlink:href="https://www.navidrome.org/docs/usage/configuration-options/"/> for supported values.
+      '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "navidrome";
+        description = "User account under which Navidrome runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "navidrome";
+        description = "Group account under which Navidrome runs.";
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    environment.etc."navidrome/settings.json".text = builtins.toJSON (cfg.settings // {
+      DataFolder = dataFolder;
+      ConfigFile = configFile;
+    } // (attrsets.optionalAttrs (attrsets.hasAttrByPath ["Port"] cfg.settings) {
+      Port = builtins.toString cfg.settings.Port;
+    }));
```

```suggestion
    environment.etc."navidrome/settings.json".text = builtins.toJSON cfg.settings;

    services.navidrome.settings = {
      DataFolder = dataFolder;
      ConfigFile = configFile;
    };
```

This is more likely what you want here, and it leaves users with an "escape hatch" to change values via `lib.mkForce` if they *really* need to.

---

I did this, the only problem of this approach is that `navidrome` wants a string as port so I cannot longer pass it as integer.

---

Looking through the [configuration format provided](https://www.navidrome.org/docs/usage/configuration-options/) it seems to accept `port` as an integer... :thinking:

---

Yes, as of 0.25.0, Navidrome accepts port as both string and integer

---

You're right, the fact is that I created this PR forking `nixpkgs` with a previous version (but now it's updated to 0.29), so I would say to keep the modifications @aanderse suggested trusting that when the PR will be merged it's going to work both with the port as a string and as an integer.
If someone has a better approach please tell me, the only alternative that came to my mind is to create a new pull request with a new fork.

---

@aciceri if you rebase your fork then you will have the most up to date version and can continue :+1: